environment-common/src/main/java/com/sequenceiq/cloudbreak/service/proxy/ProxyConfigDtoService.java

```diff
@@ -37,16 +42,43 @@ public ProxyConfig getByCrn(String resourceCrn) {
         return convert(getProxyConfig(resourceCrn, proxyEndpoint::getByResourceCrn));
     }
 
+    public Optional<ProxyConfig> getByCrnWithEnvironmentFallback(String resourceCrn, @Nonnull String environmentCrn) {
+        if (!StringUtils.isEmpty(resourceCrn)) {
+            return Optional.ofNullable(getByCrn(resourceCrn));
+        } else {
+            return getByEnvironmentCrn(environmentCrn);
+        }
+    }
+
+    public Optional<ProxyConfig> getByEnvironmentCrn(String environmentCrn) {
+        try {
+            return Optional.ofNullable(convert(proxyEndpoint.getByEnvironmentCrn(environmentCrn)));
+        } catch (NotFoundException ex) {
+            return Optional.empty();
+        }
+    }
+
+    public boolean isProxyConfiguredForEnvironment(String environmentCrn) {
+        try {
+            return proxyEndpoint.getByEnvironmentCrn(environmentCrn) != null;
+        } catch (NotFoundException ex) {
+            return false;
+        }
+    }
+
     private ProxyConfig convert(ProxyResponse proxyResponse) {
-        return ProxyConfig.builder()
+        ProxyConfig.ProxyConfigBuilder proxyConfigBuilder = ProxyConfig.builder()
                 .withName(proxyResponse.getName())
                 .withCrn(proxyResponse.getCrn())
                 .withProtocol(proxyResponse.getProtocol())
                 .withServerHost(proxyResponse.getHost())
-                .withServerPort(proxyResponse.getPort())
-                .withUserName(getSecret(proxyResponse.getUserName()))
-                .withPassword(getSecret(proxyResponse.getPassword()))
-                .build();
+                .withServerPort(proxyResponse.getPort());
+        if (proxyResponse.getUserName() != null && proxyResponse.getPassword() != null) {
```

Comment:
StringUtils#noneBlank as I think if there is faulty config (eg username is some space) in the DB from earlier or whatever, we should not use it as valid

Reply:
the username and the password is SecretResponse, not a string. Currently it is not possible to create a proxy config with nonblank username and password, we agreed that we will skip this check here.